core/src/main/scala/org/bitcoins/core/protocol/ln/LnHumanReadablePart.scala

```diff
@@ -0,0 +1,84 @@
+package org.bitcoins.core.protocol.ln
+
+import org.bitcoins.core.config.NetworkParameters
+import org.bitcoins.core.protocol.NetworkElement
+import org.bitcoins.core.protocol.ln.LnParams._
+import scodec.bits.ByteVector
+import scala.util.matching.Regex
+import scala.util.{ Failure, Try }
+
+sealed abstract class LnHumanReadablePart extends NetworkElement {
+  def network: LnParams
+
+  def amount: Option[LnCurrencyUnit]
+
+  override def toString: String = network.value + invoiceAmount
+
+  def invoiceAmount: String = {
+    if (amount.isDefined) {
+      amount.get.toBigInt.toString() + amount.get.character
+    } else ""
+  }
+
+  override def bytes: ByteVector = ByteVector(network.value.map(c => c.toByte) ++ invoiceAmount.map(c => c.toByte)) //TODO: Verify
+}
+
+case class HumanReadablePart(inNetwork: LnParams, amnt: Option[LnCurrencyUnit]) extends LnHumanReadablePart {
+  def network: LnParams = inNetwork
+  def amount: Option[LnCurrencyUnit] = amnt
+}
+
+case object lnbc extends LnHumanReadablePart {
```

Comment:
This isn't quite right, you are missing 

```scala

sealed abstract class LnPrefix 

case object lnbc extends LnPrefix  {
  def network: LnParams = MainNet 
  //note we do not want the `LnCurrencyUnit` here, put that in LnHumanReadablePart
}
```

and then `LnHumanReadablePart` should be

```scala

case class LnHumanReadablePart(prefix: LnPrefix, amount: Option[LnCurrency]

```

Reply:
I'm a bit confused on this one. A few questions.
1. Are we going with `LnPrefix` or `LnParams` as the name for the network piece?
2. I dont think the `lnbc`, `lntb`, and `lnbcrt` should extend the `LnPrefix/Params`, as that is what [these are for](https://github.com/bitcoin-s/bitcoin-s-core/blob/09014a223c98244af8f98b2fc423ed722bb3b827/core/src/main/scala/org/bitcoins/core/protocol/ln/LnParams.scala#L10)?
3. These objects exists because similar objects exist in the Bitcoin HRP. However, they are not actually be used at the moment. Should we remove them all together?

Reply:
I'm envisioning `LnParams` is similar to [`ChainParams`](https://github.com/bitcoin-s/bitcoin-s-core/blob/master/core/src/main/scala/org/bitcoins/core/protocol/blockchain/ChainParams.scala#L26) in bitcoin, which has other relevant information about the network (like ports, dns seeds, etc). The `value` field should probably be moved out of `LnParams` IMO.

Reply:
Thinking about this more, a `LnParams` should probably have a `ChainParams` val inside of it.